modules/site-health/audit-enqueued-assets/load.php

```diff
@@ -0,0 +1,304 @@
+<?php
+/**
+ * Module Name: Audit Enqueued Assets
+ * Description: Adds a CSS and JS resource checker in Site Health checks.
+ * Experimental: No
+ *
+ * @package performance-lab
+ * @since 1.0.0
+ */
+
+/**
+ * Require helper functions.
+ */
+require_once __DIR__ . '/helper.php';
+
+/**
+ * Audit enqueued scripts in is_front_page(). Ignore /wp-includes scripts.
+ *
+ * It will save information in a transient for 12 hours.
+ *
+ * @since 1.0.0
+ */
+function perflab_aea_audit_enqueued_scripts() {
+	if ( ! is_admin() && is_front_page() && current_user_can( 'view_site_health_checks' ) && false === get_transient( 'aea_enqueued_front_page_scripts' ) ) {
+		global $wp_scripts;
+		$enqueued_scripts = array();
+
+		foreach ( $wp_scripts->queue as $handle ) {
+			$src = $wp_scripts->registered[ $handle ]->src;
+			if ( $src && ! strpos( $src, 'wp-includes' ) ) {
+				$enqueued_scripts[] = array(
+					'src'  => $src,
+					'size' => perflab_aea_get_resource_file_size( perflab_aea_get_path_from_resource_url( $src ) ),
+				);
+			}
+		}
+		set_transient( 'aea_enqueued_front_page_scripts', $enqueued_scripts, 12 * HOUR_IN_SECONDS );
+	}
+}
+add_action( 'wp_print_scripts', 'perflab_aea_audit_enqueued_scripts' );
+
+/**
+ * Audit enqueued styles in the frontend. Ignore /wp-includes styles.
+ *
+ * It will save information in a transient for 12 hours.
+ *
+ * @since 1.0.0
+ */
+function perflab_aea_audit_enqueued_styles() {
+	if ( ! is_admin() && is_front_page() && current_user_can( 'view_site_health_checks' ) && false === get_transient( 'aea_enqueued_front_page_styles' ) ) {
+		global $wp_styles;
+		$enqueued_styles = array();
+		foreach ( $wp_styles->queue as $handle ) {
+			$src = $wp_styles->registered[ $handle ]->src;
+			if ( $src && ! strpos( $src, 'wp-includes' ) ) {
+				$enqueued_styles[] = array(
+					'src'  => $src,
+					'size' => perflab_aea_get_resource_file_size( perflab_aea_get_path_from_resource_url( $src ) ),
+				);
+			}
+		}
+		set_transient( 'aea_enqueued_front_page_styles', $enqueued_styles, 12 * HOUR_IN_SECONDS );
+	}
+}
+add_action( 'wp_print_styles', 'perflab_aea_audit_enqueued_styles' );
```

Comment:
A few notes for this one:

* Styles can be printed in the header or the footer. I'm not sure `wp_print_styles` would cover all scenarios
* Stylesheets can have additional styles added inline - which should also be accounted for when getting the size
* For some styles we already know their path (it was part of a refactor for block styles which was merged in 5.9)
* `$wp_styles->queue` contains styles that are to be enqueued. However, a stylesheet can be enqueued _after_  this has run, and if I'm not mistaken any styles that have already been printed will not be included here. To avoid a priorities battle and over-complicaing the implementation, it would be better if we ran this as the last action in `wp_footer`, and check which stylesheets were done.

The below code fixes the above issues:

```php

add_action( 'wp_footer', function() {
	global $wp_styles;
	$enqueued_styles = array();

	// Loop through the styles that were printed, and get their details.
	foreach ( $wp_styles->done as $style_handle ) {
		$style = $wp_styles->registered[ $style_handle ];

		if ( ! empty( $style->extra ) && ! empty( $style->extra['path'] ) ) { // Check if we already have the style's path.
			$path = $style->extra['path'];
		} else { // Fallback to getting the path from the style's src.
			$path = perflab_aea_get_path_from_resource_url( $style->src );
		}

		// Add any extra data that was passed with the style.
		$extras_size = 0;
		if ( ! empty( $style->extra ) && ! empty( $style->extra['after'] ) ) {
			foreach ( $style->extra['after'] as $extra ) {
				$extras_size += strlen( $extra );
			}
		}
		
		$enqueued_styles[] = array(
			'src' => $style->src,
			'size' => perflab_aea_get_resource_file_size( $src ) + $extras_size,
		);
	}

	set_transient( 'aea_enqueued_front_page_styles', $enqueued_styles, 12 * HOUR_IN_SECONDS );

}, PHP_INT_MAX );
```

Reply:
thanks, @aristath for this valuable feedback, would this apply also to scripts? or just to styles?

Reply:
> would this apply also to scripts? or just to styles?

Some (but not all) parts also apply to scripts, yes. Scripts can get added in the head or the footer, and they can get added while the page renders. For example, a block can enqueue its own script on-render. This will make the script get added only when the block gets rendered, but it won't be visible to `$wp_scripts` before we get to the footer.
However, scripts don't (yet) have a `path` defined, this is something we plan on doing in WP 6.0 - so that part can be skipped 👍

Reply:
Good catch @aristath! Since considering this is also adding an extra "dimension" of code to this though, based on my earlier comment, would it be okay to enhance/fix this in a follow-up issue and PR?

Reply:
Of course! We can merge this one as-is, and improve in follow-up PRs 👍